src/Bearded.TD/Game/Simulation/Elements/ArcTree.cs

```diff
@@ -0,0 +1,201 @@
+
+    ArcTree.ArcContinuation GetInitialArcParametersWithoutModifiers();
+}
+
+interface IArcModifier
+{
+    void Modify(ref ArcTree.ArcContinuation arc);
+}
+
+static class ArcTree
+{
+    private record struct ArcHit(
+        GameObject Source,
+        GameObject Target,
+        ArcContinuation Continuation
+    );
+
+    public record struct ArcContinuation(
+        GameObject Source,
+        int BouncesBefore,
+        int BouncesLeft,
+        int Branches,
+        int MaxBounceDistance)
+    {
+        public static ArcContinuation Initial(
+            GameObject source, int bounces = 1, int branches = 1, int maxBounceDistance = 2)
+            => new(source, 0, bounces, branches, maxBounceDistance);
+    }
+
+    public static void Strike(
+        GameState game,
+        GameObject? source,
+        IArcSource arcSource,
+        ImmutableArray<Tile> tilesInRange,
+        UntypedDamage damage,
+        TimeSpan lifeTime)
+    {
+        var context = setupContext(game, source, arcSource);
+
+        hitInitialTargets(context, arcSource, tilesInRange);
+
+        while (context.TryGetContinuation(out var continuation))
+        {
+            var targets = findTargetsFromContinuation(context, continuation);
+            hitTargets(context, continuation, targets);
+        }
+
+        // TODO: how should damage scale with different hit patterns
+        // variables: bounces, branches, total hits, (distance? nah)
+        // arcSource, as well as modifiers likely want to affect this
+        // could also return a list of hits, and let the caller decide for more flexibility (except YAGNI)
+        foreach (var hit in context.Arcs)
+        {
+            arc(source, hit);
+        }
+    }
+
+    private static void arc(GameObject? originalSource, ArcHit hit)
+    {
+        // hit target
+        // and create actual arc object
+    }
+
+    private static Context setupContext(GameState game, GameObject? source, IArcSource arcSource)
+    {
+        var context = new Context(
+            game,
+            game.PassabilityObserver.GetLayer(Passability.Projectile),
+            new HashSet<GameObject>(),
+            arcSource.ContinuationModifiers.ToImmutableArray(),
+            new FloodFillRanger());
+
+        if (source != null)
+            context.BlackList.Add(source);
+
+        return context;
+    }
+
+    private static void hitInitialTargets(Context context, IArcSource arcSource, ImmutableArray<Tile> tilesInRange)
+    {
+        var initialContinuation = arcSource.GetInitialArcParametersWithoutModifiers();
+        appleArcModifiers(context, ref initialContinuation);
+
+        var initialTargets = findTargetsInArea(context, initialContinuation.Branches, tilesInRange);
+        hitTargets(context, initialContinuation, initialTargets);
+    }
+
+    private static void hitTargets(Context context, ArcContinuation previousContinuation, List<GameObject> targets)
```

Comment:
Since "hitting" is already a very overloaded term in our code (and in general). It seems like what you're actually doing here is "connecting" targets to this tree, isn't that right? Would `connectTargets` or a similar name make it more obvious that this algorithm is basically creating a tree?